storage/buckets/set_bucket_public_iam.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buckets
+
+// [START storage_set_bucket_public_iam]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/iam"
+
+	"cloud.google.com/go/storage"
+)
+
+// setBucketPublicIAM makes all objects in a bucket publicly readable.
+func setBucketPublicIAM(w io.Writer, bucketName string, policy iam.Policy) error {
+	// bucketName := "bucket-name"
+	// policy := client.Bucket(bucketName).IAM().Policy(ctx)
```

Comment:
Construct the policy in the sample rather than passing it in (unless there's some reason we can't). Also, why use Policy rather than Policy3?

Reply:
@frankyn any preference on your side? I don't really understand the difference super well either...

Reply:
Please prioritize V3, we will need to update Java and C++ samples they were missed during IAM Conditions launch.

Reply:
Thanks for review, changed to V3